lib/rubocop/cop/rails/strip_heredoc.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Enforces the use of squiggly heredoc over `strip_heredoc`.
+      #
+      # @example
+      #
+      #   # bad
+      #   <<EOS.strip_heredoc
+      #     some text
+      #   EOS
+      #
+      #   # bad
+      #   <<-EOS.strip_heredoc
+      #     some text
+      #   EOS
+      #
+      #   # good
+      #   <<~EOS
+      #     some text
+      #   EOS
+      #
+      class StripHeredoc < Base
+        extend AutoCorrector
+        extend TargetRubyVersion
+
+        MSG = 'Use squiggly heredoc (`<<~`) instead of `strip_heredoc`.'
+        RESTRICT_ON_SEND = %i[strip_heredoc].freeze
+
+        minimum_target_ruby_version 2.3
+
+        def on_send(node)
+          return unless (receiver = node.receiver)
```

Comment:
Even though there's some stability in `RESTRICT_ON_SEND`, accoring to:
> The `on_send` callback is the most used and can be optimized by restricting the acceptable method names with a constant `RESTRICT_ON_SEND`.

Do you think it makes sense to keep this line just to be on the safe side?
```ruby
return unless node.method?(:strip_heredoc)
```

Reply:
My suggestion might be overly paranoid.
I can't tell the exact share, but some cops seem to have this duplication, even when there's just one short method name:

https://github.com/rubocop/rubocop/blob/cdf0d4b9abbbefdff63aab9c87c74b20bd8ed8af/lib/rubocop/cop/style/array_join.rb#L27
[lib/rubocop/cop/style/float_division.rb](https://github.com/rubocop/rubocop/blob/cdf0d4b9abbbefdff63aab9c87c74b20bd8ed8af/lib/rubocop/cop/style/float_division.rb#L68)

I'm not aware of any practice regarding this. Please don't hesitate to ignore the above suggestion completely as you see fit.

Reply:
AFAIK, `method?` may be redundant with `RESTRICT_ON_SEND` because it only checks a method name. So, it is different from the `node_matcher` that checks other than a method name.